web/search.dart

```diff
@@ -327,23 +425,45 @@ void initializeSearch(
   });
 
   input.addEventListener('keydown', (Event event) {
-    if (suggestionElements.isEmpty) {
-      return;
-    }
-
     if (event.type != 'keydown') {
       return;
     }
 
     event = event as KeyboardEvent;
 
     if (event.code == 'Enter') {
-      var selectingElement = selectedElement ?? 0;
-      var href = suggestionElements[selectingElement].dataset['href'];
-      if (href != null) {
-        window.location.assign('$htmlBase$href');
+      if (selectedElement != null) {
+        var selectingElement = selectedElement ?? 0;
+        var href = suggestionElements[selectingElement].dataset['href'];
+        if (href != null) {
+          window.location.assign('$htmlBase$href');
+        }
+        return;
+      }
+      // If there no search suggestion selected then change the window location to the search.html.
+      if (selectedElement == null ||
+          listBox.getAttribute('aria-expanded') == 'true' ||
+          suggestionElements.isEmpty) {
+        // Saves the input in the search to be used for creating the query parameter.
+        var input = htmlEscape.convert(actualValue);
+        var relativePath = '';
+        if (document
+                    .querySelector('body')
```

Comment:
Let's save `document.querySelector('body')` as a local variable. This should make the code below shorter, and avoid multiple null-checks and null-asserts.